src/parser/program.rs

```diff
@@ -0,0 +1,54 @@
+use serde::{Deserialize};
+use serde_yaml::Value;
+use std::{collections::HashMap, fmt::Display};
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pub programs: HashMap<String, Program>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Program {
+    name: Option<String>,
+    pid: Option<u32>,
+    cmd: Option<String>,
+    numprocs: Option<u32>,
+    workingdir: Option<String>,
+    autostart: Option<bool>,
+    exitcodes: Option<Vec<u8>>, // check for valid codes (%256)
+    startretries: Option<u32>,
+    starttime: Option<u32>,
+    stopsignal: Option<String>, // check for valid signal
+    stoptime: Option<u32>,
+    stdout: Option<String>,
+    stderr: Option<String>,
+    env: Option<HashMap<String, Value>>,
+}
```

Comment:
None of those should be optional.
Make 2 structs, one that is used only during parsing with Option when needed (ie: name) and one for the fully parsed Program that has no Option.